src/arrayMethodSort.js

```diff
@@ -5,7 +5,31 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    if (compareFunction) {
+      for (let i = 0; i < this.length - 1; i++) {
+        for (let j = 0; j < this.length - 1 - i; j++) {
+          if (compareFunction(this[j], this[j + 1]) > 0) {
+            const temp = this[j];
+
+            this[j] = this[j + 1];
+            this[j + 1] = temp;
+          }
+        }
+      }
+    } else {
+      for (let i = 0; i < this.length - 1; i++) {
+        for (let j = 0; j < this.length - 1 - i; j++) {
+          if (String(this[j]) > String(this[j + 1])) {
+            const temp = this[j];
+
+            this[j] = this[j + 1];
+            this[j + 1] = temp;
+          }
+        }
+      }
+    }
```

Comment:
Now this code is practically duplicated, try to create a helper function at the top level and pass it as a function parameter, it will simplify your code